### PR TITLE
CP-43266: New notifications page and alerts for out of sync with repo

### PR DESCRIPTION
**Pull request by kc284**
NOTE: please review all the other commits except the first one titled "DROP ME..." because this was added already in https://github.com/xenserver/xenadmin/pull/3163 and has been reviewed, the plan is to merge 3163 first and then rebase this PR on top, so it will be dropped.

Known issue with the last commit titled "CP-43266...": the autoresizing of the expandable column that lists the updates needs reworking because it is a custom drawn DataGridViewTextBoxCell and the datagridview seems to take account only of the text width, hence the bottom of the cells is sometimes cropped. I'll rework this in a future PR.

### CODE REVIEW

File: XenAdmin/Alerts/Types/OutOfSyncWithCdnAlert.cs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                {
+                    using (var dialog = new ConfigUpdatesDialog())
+                        dialog.ShowDialog(Program.MainWindow);
+                };
+            }
+        }
+
+        public override string FixLinkText => Messages.ALERT_CDN_REPO_NOT_CONFIGURED_ACTION_LINK;
+
+        public override string HelpID => "TODO";
```
Comment (CitrixChris) [inline XenAdmin/Alerts/Types/OutOfSyncWithCdnAlert.cs:155]: There is a TODO here, shouldnt this have a value now?

File: XenAdmin/Alerts/Types/OutOfSyncWithCdnAlert.cs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                    var syncAction = new SyncWithCdnAction(_pool);
+                    syncAction.Completed += a => Updates.CheckForCdnUpdates(a.Connection);
+                    syncAction.RunAsync();
+                };
+            }
+        }
+
+        public override string FixLinkText => Messages.UPDATES_GENERAL_TAB_SYNC_NOW;
+
+        public override string HelpID => "TODO";
```
Comment (CitrixChris) [inline XenAdmin/Alerts/Types/OutOfSyncWithCdnAlert.cs:103]: TODO value, shouldnt this have a value now?

Comment (kc284) [reply XenAdmin/Alerts/Types/OutOfSyncWithCdnAlert.cs:103]: No help IDs are known yet.

**Comment by kc284** [comment]
I rebased this PR on master, so the only commis for review here are:

[Show update status for pools and hosts on the General tabPage:](https://github.com/xenserver/xenadmin/pull/3182/commits/ddf20f6c417e92312706157f2332880f015ed557) 
[ddf20f6](https://github.com/xenserver/xenadmin/pull/3182/commits/ddf20f6c417e92312706157f2332880f015ed557)

[Alert the user if they haven't synced in a long time. Also:](https://github.com/xenserver/xenadmin/pull/3182/commits/bdffc1d2cc25b3367e9eab3a12dfde03f827a7ac) 
[bdffc1d](https://github.com/xenserver/xenadmin/pull/3182/commits/bdffc1d2cc25b3367e9eab3a12dfde03f827a7ac)

[CP-43266: Added new notifications page to list updates from CDN.](https://github.com/xenserver/xenadmin/pull/3182/commits/ea2d1a0189404e4d3db7365b1ddf462bfba9d522) 
[ea2d1a0](https://github.com/xenserver/xenadmin/pull/3182/commits/ea2d1a0189404e4d3db7365b1ddf462bfba9d522)

File: XenAdmin/TabPages/CdnUpdates/CdnExpandableRow.cs [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+                if (Host.software_version.ContainsKey("date"))
+                {
+                    if (!Util.TryParseIso8601DateTime(Host.software_version["date"], out softwareVersionDate))
+                        Util.TryParseNonIso8601DateTime(Host.software_version["date"], out softwareVersionDate);
+                }
+
+                if (Host.last_software_update > softwareVersionDate && Host.last_software_update > unixMinDateTime)
+                {
+                    lastUpdateTime = HelpersGUI.DateTimeToString(Host.last_software_update.ToLocalTime(), Messages.DATEFORMAT_DMY_HMS, true);
```
Comment (danilo-delbusso) [inline XenAdmin/TabPages/CdnUpdates/CdnExpandableRow.cs:151]: Can't we try to use the `Pool.last_update_sync.ToLocalTime()` from the host's pool?

If it's not possible then let's leave it empty but I can go to the host's General Tab and see last sync there, and then it doesn't show here.

Comment (danilo-delbusso) [reply XenAdmin/TabPages/CdnUpdates/CdnExpandableRow.cs:151]: I understand we might not want to show it for every row in a pool with several hosts. So maybe only show if the host is standalone?

**Review by danilo-delbusso — CHANGES_REQUESTED** [review]
small tweak.

Also should we rename the `Synchronize` button in the stream updates page to `Synchronize selected`? Wasn't 100% clear to me at first that it was just the one I had selected

**Review by CitrixChris — APPROVED** [review]

**Review by danilo-delbusso — APPROVED** [review]

**Review by neatleygit — APPROVED** [review]

**Review by CitrixChris — APPROVED** [review]

**Pull request merged**